plans/types.ts

```
@@ -0,0 +1,32 @@
+  C extends BaseContext<P>,
+> = (context: C) => Promise<Result<C>>;
+
+export interface Step<P extends Record<string, any>, C extends BaseContext<P>> {
+  name: string;
+  prompt: (context: C) => string;
+  guard?: StepGuard<P, C>;
+}
+
+export interface SimpleWorkflow<
+  P extends Record<string, any>,
+  C extends Record<string, any> = {},
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)